src/Components/Blazor/Blazor/src/Services/WebAssemblyNavigationManager.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+using Interop = Microsoft.AspNetCore.Components.Web.BrowserNavigationManagerInterop;
+
+namespace Microsoft.AspNetCore.Blazor.Services
+{
+    /// <summary>
+    /// Default client-side implementation of <see cref="NavigationManager"/>.
+    /// </summary>
+    public class WebAssemblyNavigationManager : NavigationManager
+    {
+        /// <summary>
+        /// Gets the instance of <see cref="WebAssemblyNavigationManager"/>.
+        /// </summary>
+        public static readonly WebAssemblyNavigationManager Instance = new WebAssemblyNavigationManager();
```

Comment:
If there is this public singleton, people will use it instead of the DI service. Are we OK with that?

I'm not trying to force ceremony on people just for the sake of it, but if people use `WebAssemblyNavigationManager.Instance` then they immediately lost portability between client and server Blazor for that component.

Reply:
I think we should avoid the public instance if we can for the reasons @SteveSandersonMS mentions.

Reply:
Is there a good reason for this type to be public? Presumably we want users to program against the `NavigationManager` instance?

Reply:
It's required right now because of JSInterop used in the implementation. I was hoping to avoid too much polish here because it's in the Blazor client code.

I'm happy to make the field internal.

Reply:
Internal sounds good.